src/dfx/src/lib/network/local_server_descriptor.rs

```diff
@@ -0,0 +1,17 @@
+use crate::config::dfinity::to_socket_addr;
+use crate::lib::error::DfxResult;
+
+use fn_error_context::context;
+use std::net::SocketAddr;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct LocalServerDescriptor {
+    pub bind: String,
+}
+
+impl LocalServerDescriptor {
+    #[context("Failed to convert {} to a bind address", self.bind)]
+    pub fn bind_address(&self) -> DfxResult<SocketAddr> {
+        to_socket_addr(&self.bind)
+    }
```

Comment:
@sesi200 The "get" happens only one time, in `dfx start` when calculating the frontend address, so I left it this way so that it only happens there instead of on every dfx command.

This is the equivalent of `Environment::get_local_bind_address` in the previous code.

Reply:
@sesi200 The "get" happens only one time, in `dfx start` when calculating the frontend address, so I left it this way so that it only happens there instead of on every dfx command.

It's the equivalent of `Environment::get_local_bind_address` in the previous code.

But this is something I thought about too, so if someone else is asking as well, it probably makes sense to go the other way.